lib/async/enumerable.rb

```diff
@@ -22,15 +23,26 @@ module Async
   # Provides async parallel execution for Enumerable.
   # See docs/reference/enumerable.md for detailed documentation.
   module Enumerable
-    DEFAULT_MAX_FIBERS = 1024
-
     class << self
-      attr_writer :max_fibers
+      # Gets or creates the module-level config.
+      # @param kwargs [Hash] Optional configuration updates
+      # @return [Config] Module configuration
+      def config(**kwargs)
+        @config ||= Config.default
+        @config = @config.with(**kwargs) unless kwargs.empty?
+        @config
+      end
```

Comment:
i'm concerned about concurrency safety here -- since this is a singleton, it's very likely to be accessed concurrently. not sure what to do about it yet.